src/main/java/ch/njol/skript/sections/EffSecSpawn.java

```diff
@@ -161,15 +163,26 @@ protected TriggerItem walk(Event event) {
 		Number numberAmount = amount != null ? amount.getSingle(event) : 1;
 		if (numberAmount != null) {
 			double amount = numberAmount.doubleValue();
-			EntityType[] types = this.types.getArray(event);
+			Object[] types = this.types.getArray(event);
 			for (Location location : locations.getArray(event)) {
-				for (EntityType type : types) {
-					double typeAmount = amount * type.getAmount();
-					for (int i = 0; i < typeAmount; i++) {
-						if (consumer != null) {
-							type.data.spawn(location, (Consumer) consumer); // lastSpawned set within Consumer
-						} else {
-							lastSpawned = type.data.spawn(location);
+				for (Object type : types) {
+					if (type instanceof EntityType entityType) {
+						double typeAmount = amount * entityType.getAmount();
+						for (int i = 0; i < typeAmount; i++) {
+							if (consumer != null) {
+								entityType.data.spawn(location, (Consumer) consumer); // lastSpawned set within Consumer
+							} else {
+								lastSpawned = entityType.data.spawn(location);
+							}
+						}
+					} else if (type instanceof EntitySnapshot snapshot) {
+						for (int i = 0; i < amount; i++) {
+							Entity entity = snapshot.createEntity(location);
+							if (consumer != null) {
+								((Consumer<Entity>) consumer).accept(entity);
```

Comment:
we should note that snapshots won't have their consumers executed quite as fast as entity types in the description.